src/app/tvshows/store/tvshows.actions.ts

```diff
@@ -0,0 +1,58 @@
+import { Action } from 'redux';
+import { TvShowData } from './tvshows.models';
+import { TvDbService } from '../../services';
+
+/**
+ * Actions key as enum types
+ *
+ * Typescripts's enum is ideal for this purpose, vs constants.  Plus
+ * we can use each enum value as a type specifier.  Combined with 'readonly' ensures
+ * actions keys are immutable and contain the intended value.
+ */
+
+export enum TypeKeys {
+  SEARCH_SHOWS = 'tvshows/search',
+  SEARCH_SHOWS_SUCCESS = 'tvshows/search success',
+  SEARCH_SHOWS_FAILURE = 'tvshows/search failure',
+}
+
+export interface TvShowSearchSuccess extends Action {
+  readonly type: TypeKeys.SEARCH_SHOWS_SUCCESS;
+  by: TvShowData[];
+}
+
+export const searchSuccess = (by: TvShowData[]): TvShowSearchSuccess => ({
+  type: TypeKeys.SEARCH_SHOWS_SUCCESS,
+  by,
+});
+
+export interface TvShowSearchShows extends Action {
+  readonly type: TypeKeys.SEARCH_SHOWS;
+  by: string;
+}
+
+
+// Thunk style action creator, return type is 'any' otherwise typescript IDEA
+// will compain its not of type Action:
+export function searchShows(query: string, tvdb: TvDbService): any {
```

Comment:
I haven't used thunks, but it seems odd that you pass in the tv service here. It basically combines 2 different concepts (action creator & side effect handler). If that's the norm, then I'm personally not a big fan of this pattern.

Reply:
Yes, in thunks the action creator performs the side-effect.  So it needs the tvdb service to perform the API call.